src/python/pants/base/exception_sink.py

```diff
@@ -445,6 +445,12 @@ def _log_unhandled_exception_and_exit(cls, exc_class=None, exc=None, tb=None, ad
       should_print_backtrace=cls._should_print_backtrace_to_terminal)
     if extra_err_msg:
       stderr_printed_error = '{}\n{}'.format(stderr_printed_error, extra_err_msg)
+
+    return stderr_printed_error
+
+  @classmethod
+  def _log_unhandled_exception_and_exit(cls, exc_class=None, exc=None, tb=None, add_newline=False):
```

Comment:
Because this is a new method, I recommend adding kwarg only arguments:

```python
 def _log_unhandled_exception_and_exit(cls, *, exc_class=None, exc=None, tb=None, add_newline=False):
```

Maybe not necessary for first 3 args, but definitely would use for `add_newline`.

Would be great to use the above type hints here too :) Here, the return type is `from typing import NoReturn` because it's guaranteed this will never return.